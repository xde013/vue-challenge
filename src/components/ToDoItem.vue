<template>
  <div class="todo-item">

    <input type='checkbox' v-model="todo.checked">
     {{this.todo.text}}
     <span class='timeago'> - 5 minutes ago</span>
  </div>
</template>

<script>
export default {
  name: 'ToDoItem',
  props: {
    todo: {
      required: true
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus">
    .timeago {
        opacity .5
        font-size .8em
        font-weight 200
    }
</style>
