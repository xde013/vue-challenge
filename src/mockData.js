export const todos = [
  {
    text: 'Think real hard about whats for lunch',
    checked: false,
    createdAt: new Date(Date.now() - 3600)
  },
  {
    text: 'Buy some Pantos and start looking for Lambos',
    checked: false,
    createdAt: new Date(Date.now() - 7200)
  }
]
