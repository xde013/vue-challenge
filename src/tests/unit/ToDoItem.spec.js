import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import ToDoItem from '@/components/ToDoItem.vue'
import { todos } from '@/mockData'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('ToDoItem.vue', () => {
  it('renders props.todo when passed', () => {
    const todo = todos[0]
    const wrapper = shallowMount(ToDoItem, {propsData: { todo }})
    expect(wrapper.props().todo).toBe(todo)
    expect(wrapper.find('.todo-text').text()).toBe(todo.text)
  })
  it('matches snapshot', () => {
    const wrapper = shallowMount(ToDoItem, {propsData: {todo: todos[0]}})
    expect(wrapper.html()).toMatchSnapshot()
  })
})
