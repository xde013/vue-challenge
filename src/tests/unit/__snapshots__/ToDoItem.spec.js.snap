// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoItem.vue matches snapshot 1`] = `
<div class="todo-item">
  <input type="checkbox"> <span class="todo-text"> Think real hard about whats for lunch </span> <span class="timeago"> 3 seconds ago </span>
  <button class="remove"> X </button>
</div>
`;
